flink-tests/src/test/java/org/apache/flink/test/javaApiOperators/PartitionITCase.java

```diff
@@ -546,43 +549,274 @@ public void testRangePartitionInIteration() throws Exception {
 		result.collect(); // should fail
 	}
 
+
+
+	@Test
+	public void testRangePartitionerOnSequenceDataWithOrders() throws Exception {
+		final ExecutionEnvironment env = ExecutionEnvironment.getExecutionEnvironment();
+		DataSet<Tuple2<Long, Long>> dataSet = env.generateSequence(0, 10000)
+				.map(new MapFunction<Long, Tuple2<Long, Long>>() {
+			@Override
+			public Tuple2<Long, Long> map(Long value) throws Exception {
+				return new Tuple2<>(value / 5000, value % 5000);
+			}
+		});
+
+		final Tuple2Comparator<Long> tuple2Comparator = new Tuple2Comparator<>(new LongComparator(true),
+																			   new LongComparator(false));
+
+		MinMaxSelector<Tuple2<Long, Long>> minMaxSelector = new MinMaxSelector<>(tuple2Comparator);
+
+		final List<Tuple2<Tuple2<Long, Long>, Tuple2<Long, Long>>> collected = dataSet.partitionByRange(0, 1)
+				.withOrders(Order.ASCENDING, Order.DESCENDING)
+				.mapPartition(minMaxSelector)
+				.collect();
+
+		Collections.sort(collected, new Tuple2Comparator<>(tuple2Comparator));
```

Comment:
Maybe a duplicate response, but it is only applied in sorting. The overlapping will be checked in prevMax < curMin. 

Of course I could add some comparator that will throw exception on comparing the second element, but I think it will only complicate the readability.